packages/payment-processor/src/payment/utils.ts

```diff
@@ -111,7 +116,10 @@ export function validateRequest(
     !extension ||
     !extension.values.salt ||
     !extension.values.paymentAddress ||
-    (paymentNetworkId === ERC20_PROXY_CONTRACT && !request.currencyInfo.value)
+    (paymentNetworkId === ERC20_PROXY_CONTRACT && !request.currencyInfo.value) ||
+    (paymentNetworkId === ERC20_FEE_PROXY_CONTRACT && !request.currencyInfo.value) ||
+    (paymentNetworkId === ERC20_FEE_PROXY_CONTRACT &&
+      !!extension.values.feeAddress !== !!extension.values.feeAmount)
```

Comment:
aren't those 2 mandatory for an `ERC20_FEE_PROXY_CONTRACT`?

Reply:
no, it's mandatory to have both, or neither. If neither, it just behaves like a normal erc20 proxy.